app-crypt/argon2/patches/argon2-20171227.patchset

```diff
@@ -0,0 +1,78 @@
+From 2e078b7a9903f96377a1f7c59f95182518204895 Mon Sep 17 00:00:00 2001
+From: Deposite Pirate <dpirate@metalpunks.info>
+Date: Thu, 13 Sep 2018 21:33:03 +0200
+Subject: Fix building on Haiku
+
+
+diff --git a/Makefile b/Makefile
+index 1e1129f..ea97411 100644
+--- a/Makefile
++++ b/Makefile
+@@ -30,13 +30,18 @@ SRC_BENCH = src/bench.c
+ SRC_GENKAT = src/genkat.c
+ OBJ = $(SRC:.c=.o)
+ 
++BUILD_PATH := $(shell pwd)
++KERNEL_NAME := $(shell uname -s)
++
+ CFLAGS += -std=c89 -O3 -Wall -g -Iinclude -Isrc
+ 
+ ifeq ($(NO_THREADS), 1)
+ CFLAGS += -DARGON2_NO_THREADS
+ else
++ifneq ($(KERNEL_NAME), Haiku)
+ CFLAGS += -pthread
+ endif
++endif
+ 
+ CI_CFLAGS := $(CFLAGS) -Werror=declaration-after-statement -D_FORTIFY_SOURCE=2 \
+ 				-Wextra -Wno-type-limits -Werror -coverage -DTEST_LARGE_RAM
+@@ -54,9 +59,6 @@ $(info Building with optimizations for $(OPTTARGET))
+ 	SRC += src/opt.c
+ endif
+ 
+-BUILD_PATH := $(shell pwd)
+-KERNEL_NAME := $(shell uname -s)
+-
+ LIB_NAME=argon2
+ ifeq ($(KERNEL_NAME), Linux)
+ 	LIB_EXT := so.$(ABI_VERSION)
+@@ -91,6 +93,12 @@ ifeq ($(KERNEL_NAME), SunOS)
+ 	LIB_EXT := so
+ 	LIB_CFLAGS := -shared -fPIC
+ endif
++ifeq ($(KERNEL_NAME), Haiku)
++	LIB_EXT := so.$(ABI_VERSION)
++	LIB_CFLAGS := -shared -fPIC
++	CFLAGS += -D_REENTRANT
```

Comment:
```suggestion
+	SO_LDFLAGS := -Wl,-soname,lib$(LIB_NAME).$(LIB_EXT)
```

We need this for the soname to be generated. Also `-D_REENTRANT` doesn't mean anything on Haiku and it's not used by this library, so we should remove that.